src/main/java/org/example/bookshop/service/impl/BookServiceImpl.java

```diff
@@ -3,51 +3,51 @@
 import java.util.List;
 import lombok.RequiredArgsConstructor;
 import org.example.bookshop.dto.BookDto;
+import org.example.bookshop.dto.BookDtoWithoutCategoryIds;
 import org.example.bookshop.dto.CreateBookRequestDto;
 import org.example.bookshop.exception.EntityNotFoundException;
 import org.example.bookshop.mapper.BookMapper;
 import org.example.bookshop.model.Book;
 import org.example.bookshop.repository.BookRepository;
+import org.example.bookshop.repository.CategoryRepository;
 import org.example.bookshop.service.BookService;
 import org.springframework.data.domain.Pageable;
+import org.springframework.data.domain.Sort;
 import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
 
 @RequiredArgsConstructor
 @Service
 public class BookServiceImpl implements BookService {
     private final BookRepository bookRepository;
+    private final CategoryRepository categoryRepository;
     private final BookMapper bookMapper;
 
     @Override
     public BookDto save(CreateBookRequestDto requestDto) {
-        Book book = bookMapper.toModel(requestDto);
-        return bookMapper.toDto(bookRepository.save(book));
+        return bookMapper.toDto(bookRepository.save(bookMapper.toModel(requestDto)));
     }
 
     @Override
+    @Transactional
     public BookDto findById(Long id) {
         Book book = bookRepository.findById(id).orElseThrow(
                 () -> new EntityNotFoundException("Can't find book by id: " + id));
         return bookMapper.toDto(book);
     }
 
     @Override
-    public List<BookDto> findAll(Pageable pageable) {
+    @Transactional
```

Comment:
remove annotation

Reply:
but there is LazyInitializationException without it annotation, because of I need to return categories.
I don't need to use FetchType.EAGER, so how can I load categories, by @Query annotation?